beamer/state_machine.py

```diff
@@ -329,9 +329,13 @@ def _handle_request_resolved(event: RequestResolved, context: Context) -> Handle
     return True, None
 
 
-def _handle_hash_invalidated(event: HashInvalidated, context: Context) -> HandlerResult:
+def _handle_fill_invalidated(event: FillInvalidated, context: Context) -> HandlerResult:
     fill_block = context.fill_manager.web3.eth.get_block(event.block_number)
-    claims = _find_claims_by_fill_hash(context, event.fill_hash)
+    # FIXME: Remove with https://github.com/beamer-bridge/beamer/issues/994
+    request = context.requests.get(event.request_id)
+    assert request is not None
+
+    claims = _find_claims_by_fill_hash(context, request.fill_hash_with_fill_id(event.fill_id))
```

Comment:
This PR is said to resolve #993 and #994 so I'm wondering why we have this here.